### PR TITLE
Register and login api [TAS-40]

### PR DESCRIPTION
**Pull request by femacedogomes**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,5 @@
 node_modules
-.next
\ No newline at end of file
+.next
+
+# dot env
+*.env
```
Comment (renanpontez) [inline .gitignore:5]: Change this to .env*

Comment (renanpontez) [reply .gitignore:5]: Don't forget to add a new line at the end of ALL the files

File: src/config/config.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export default {
+  API_URL: process.env.NEXT_PUBLIC_API_URL,
+}
```
Comment (renanpontez) [inline src/config/config.ts:3]: Don't forget to add a new line at the end of ALL the files

File: .env [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# Comunication With Server
+NEXT_PUBLIC_API_URL=https://ez2live-dev-c781db305a07.herokuapp.com/v1
+
+# DATABASE
+MONGODB_URL=mongodb+srv://dbuser:0KEOy46rIJJ4Pq3D@dev-cluster.mzkybvz.mongodb.net/?retryWrites=true&w=majority
```
Comment (renanpontez) [inline .env:5]: Please remove this. MongoDB URL is not need in the front-end. Also the password is being leaked.

File: src/types/auth/index.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+export interface LogInType {
+  email:string;
```
Comment (renanpontez) [inline src/types/auth/index.ts:2]: add space

File: src/types/auth/index.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+export interface LogInType {
```
Comment (renanpontez) [inline src/types/auth/index.ts:1]: Rename this to: `ILogIn ` to follow the best practice

File: src/types/auth/index.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+  email:string;
+  password: string;
+};
+
+export interface RegisterAccountType {
```
Comment (renanpontez) [inline src/types/auth/index.ts:6]: Rename this to: `IRegisterAccount` to follow the best practice

File: src/service/auth/service.ts [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+const UserService = {
+  Login: async (data:LogInType) => {
+    const response = await http.POST('auth/login', data);
+    return response;
+  },
+  Register: async (data: RegisterAccountType) => {
+    const response = await http.POST('auth/register', data);
+    return response;
+  },
+}
```
Comment (renanpontez) [inline src/service/auth/service.ts:13]: Let's extract the login and register functions to outside this UserService object which is not needed.

We need to export only the functions directly.

**Review by renanpontez — CHANGES_REQUESTED** [review]
Left some comments. Also @pedrohbl03 will take a look

**Review by pedrohbl03 — APPROVED** [review]

**Pull request merged**